WcaOnRails/app/views/competitions/_competition_info.html.erb

```diff
@@ -171,22 +171,23 @@
         </div>
       </dd>
       <% if competition.results_posted? %>
-        <dt><%= t '.highlights' %></dt>
-        <dd>
-          <div id="show_highlights">
-            <%= t('.click_to_display_highlights_html', link_here: link_to(t('common.here'), "#")) %>
-          </div>
-          <div id="hide_highlights">
-            <%= link_to(t('.hide_highlights'), "#") %>
-          </div>
-          <div class="collapse" id="highlights_text">
-            <% if main_event.present? %>
-              <p><%=md winners(competition, main_event) %></p>
-            <% end %>
-
-            <p><%=md records(competition) %></p>
-          </div>
-        </dd>
+        <% records = md records(competition) %>
+        <% main_event_message = main_event.present? ? (md winners(competition, main_event)) : "" %>
+        <% if !main_event_message.blank? || !records.blank? %>
```

Comment:
Here you can keep the markdown rendering within the block by doing something like:
```suggestion
        <% records = records(competition) %>
        <% if main_event.present? || records.present? %>
```
and below
```erb
              <p><%= md(winners(competition, main_event)) if main_event.present? %></p>
              <p><%= md(records) %></p>
```